weldx/core.py

```diff
@@ -822,22 +822,64 @@ class GenericSeries:
     """Describes a quantity depending on one or more parameters."""
 
     _allowed_variables: list[str] = []
-    """Allowed variable names"""
+    """A list of allowed variable names.
+
+    The names are only relevant if the created `GenericSeries` is based on expressions.
+    If the expression contains any other variable name that is not part of the list an,
+    an exception is raised. It is not required that an expression includes all these
+    variables. Additionally, the expression can contain other symbols if they are used
+    as parameters.
+    """
     _required_variables: list[str] = []
-    """Required variable names"""
+    """A list of required variable names.
+
+    The names are only relevant if the created `GenericSeries` is based on expressions.
+    If one or more variables are missing in the expression, an exceptions is raised.
+    Note, that the required symbols must be variables of the expression. Using one or
+    more as a parameter will also trigger an exception.
+    """
 
     _evaluation_preprocessor: dict[str, Callable] = {}
-    """Function that should be used to adjust a var. input - (f.e. convert to Time)"""
+    """A dictionary of preprocessor functions for input variables during evaluation.
+
+    When evaluating a `GenericSeries`, the passed keyword arguments are checked against
+    the dictionaries keys. If a match is found, the corresponding preprocessor function
+    is called with the variables value and returns the updated value. As an example,
+    this can be used to support multiple time formats. The key might be ``t`` and the
+    preprocessor function would turn the original time data into an equivalent
+    `xarray.DataArray`.
+    """
 
     _required_dimensions: list[str] = []
-    """Required dimensions"""
+    """A list of required dimension names.
+
+    Explicit `GenericSeries` need all of the listed dimensions. Otherwise an exception
+    is raised. If the series is based on an expression, the dimension can either be
+    represented by a variable or be part of one of the expressions parameters.
+    """
+
     _required_dimension_units: dict[str, pint.Unit] = {}
-    """Required units of a dimension"""
+    """A dictionary that maps required units to a dimension.
+
+    If a dimension matches one of the keys of this dictionary, it is checked that the
+    units are identical to the requirement. If not, an exception is raised.
+    """
```

Comment:
```suggestion
    """A dictionary that maps required unit dimensionality to a dimension.

    If a dimension matches one of the keys of this dictionary, its dimensionality
    is checked against the requirement listed.
    """
```
Do we check against exact units or dimensionality here @vhirtham ?
If units then disregard my suggestions.

Reply:
Dimensionality